blessed/sequences.py

```diff
@@ -219,8 +227,15 @@ def _handle_long_word(self, reversed_chunks, cur_line, cur_len, width):
             idx = nxt = 0
             for text, _ in iter_parse(term, chunk):
                 nxt += len(text)
-                if Sequence(chunk[:nxt], term).length() > space_left:
-                    break
+                seq_length = Sequence(chunk[:nxt], term).length()
```

Comment:
when seq_length is equal to space_left, the for loop does the next chunk even though there is no space remaining

is there another small optimization possible to call length fewer times? something like:
```
if seq_length == space_left:
    idx = nxt
    break
elif seq_length > space_left:
    ...
````

Reply:
There are many unicode characters that consume 0 cells, https://wcwidth.readthedocs.io/en/latest/specs.html#width-of-0, there are examples of this in the ZWJ test added, though it gets it wrong in that specific case, we would prefer to cojoin with 1,800+ possible Combining Marks that might follow, like in `cafe\u0301` we would not wish to break immediately after the letter 'e', because the next character `\u0301` is desired

Reply:
Added test in 6beb3c67fe221150f47cbe9b6c9bae332af81520 hope that makes it more clear